pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -542,6 +542,30 @@ func (facts *PipelineRunFacts) GetPipelineTaskStatus() map[string]string {
 	return tStatus
 }
 
+// GetPipelineTaskStatus returns the status of a PipelineFinalTask depending on its taskRun
+func (facts *PipelineRunFacts) GetPipelineFinalTaskStatus() map[string]string {
+	// construct a map of tasks.<pipelineTask>.status and its state
```

Comment:
Looks like this code is very similar to `GetPipelineTaskStatus`. Can we refactor this part :pray

Reply:
just the switch part is overlapping, I think that's fine? 🤔